File: SteamDroid/src/com/flyingpie/steamdroid/app/Friends.java
package com.flyingpie.steamdroid.app;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ListAdapter;
import android.widget.ListView;

import com.flyingpie.steamdroid.R;
import com.flyingpie.steamdroid.api.Friend;
import com.flyingpie.steamdroid.util.SteamAdapters;
import com.flyingpie.steamdroid.util.SteamAlerts;

public class Friends extends ListActivity {
	
	private ListAdapter adapter;
	
	/** Called when the activity is first created. */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        adapter = SteamAdapters.getFriendsAdapter();
        setListAdapter(adapter);
        getListView().setBackgroundColor(getResources().getColor(R.color.listBackground));
    }
    
	@Override
	protected void onListItemClick(ListView l, View v, int position, long id) {
		super.onListItemClick(l, v, position, id);
		
		Friend friend = (Friend)adapter.getItem(position);
		
		if(friend.isAvailable() || SteamAdapters.getChatAdapter(friend).getCount() > 0)
		{			
			Intent intent = new Intent(Friends.this, Chat.class);
			intent.putExtra("steamid", friend.getSteamId());
			startActivity(intent);
		}
		else
		{
			SteamAlerts.showAlert("Warning", friend.getName() + " is offline", this);
		}
	}
}

File: SteamDroid2/SteamDroid2/App/Friends.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Views;
using Android.Widget;
using SteamDroid.Adapters;
using SteamDroid.Api;
using SteamDroid.Util;

namespace SteamDroid.App
{
    [Activity (Label = "Friends")]			
    public class Friends : ListActivity
    {
        protected override void OnCreate (Bundle bundle)
        {
            base.OnCreate (bundle);
            
            ListAdapter = SteamAdapters.GetFriendsAdapter();
            ListView.SetBackgroundColor(Resources.GetColor(Resource.Color.ListBackground));
        }
        
        protected override void OnListItemClick (ListView l, View v, int position, long id)
        {
            base.OnListItemClick (l, v, position, id);
            
            FriendsAdapter adapter = SteamAdapters.GetFriendsAdapter();
            Friend friend = adapter.GetFriendAt(position);

            if (SteamService.GetClient().Friends.GetPersonaState() == SteamKit2.EPersonaState.Offline)
            {
                SteamAlerts.ShowToast("Your state is set to offline");
            }
            else if (friend.State == SteamKit2.EPersonaState.Offline)
            {
                SteamAlerts.ShowToast(friend.Name + " is offline");
            }
            else
            {
                Intent chatIntent = new Intent(this, typeof(Chat));
                chatIntent.PutExtra("steam_id", friend.SteamId.ToString());
                StartActivity(chatIntent);
            }
        }
    }
}

